api/post.go

```diff
@@ -705,7 +705,7 @@ func sendNotifications(c *Context, post *model.Post, team *model.Team, channel *
 			}
 
 			if userAllowsEmails && status.Status != model.STATUS_ONLINE {
-				go sendNotificationEmail(c, post, profileMap[id], channel, team, senderName[id], sender)
+				sendNotificationEmail(c, post, profileMap[id], channel, team, senderName[id], sender)
```

Comment:
yes, I do remember adding the go routine cause it was causing the delays posting while waiting for the email to be sent (in case there are a bunch of emails to be sent it can hang for a while)

Maybe we should consider having the `sendNotificationEmail` and `sendPushNotifications` to receive the whole profileMap as arguments and send them in the background